### ISSUE TITLE
Cannot use ansible.utils.vars.load_options_vars provided vars in tags

### ISSUE THREAD

**Issue by willthames** [ansible/ansible#42492]
##### SUMMARY
With the changes to how tags interact with include_roles etc., there seems to be a need to pass the tags passed through to the role

The problem here is that include_role converts the contents of `tags` to a list, before templating it.

So although `tags: {{ ansible_run_tags }}` should be converted to a list of tags, it's converted to a list containing the template - later templating converts it to a list of lists. Either the `list` of `list` should be detected, or the templating should happen earlier with the role include code.

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible-playbook 2.6.0
  config file = None
  configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /Users/will/src/opensource/ansible/bin/ansible-playbook
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
No config file found; using defaults

```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE

runtags.yml
```
- hosts: localhost

  tasks:
    - include_role:
        name: common
      tags: "{{ ansible_run_tags | default('always') }}"
```

roles/common/tasks/main.yml
```
- block:
  - name: Try mytag task
    debug:
      msg: my tag
  tags: mytag

- block:
  - name: Try anothertag task
    debug:
      msg: another tag
  tags: anothertag
```

##### EXPECTED RESULTS
Ansible would run the tasks in common tagged with the tags passed on the command line

##### ACTUAL RESULTS

```
$ ansible-playbook runtags.yml -t mytag -vvv
ansible-playbook 2.6.0
  config file = None
  configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /Users/will/src/opensource/ansible/bin/ansible-playbook
  python version = 2.7.15 (default, Jun 17 2018, 12:46:58) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
No config file found; using defaults
 [WARNING]: Unable to parse /etc/ansible/hosts as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAYBOOK: runtags.yml *************************************************************************************************************************************************************************************************************************
1 plays in runtags.yml


PLAY [localhost] ******************************************************************************************************************************************************************************************************************************
ERROR! Unexpected Exception, this is probably a bug: unhashable type: 'list'
the full traceback was:

Traceback (most recent call last):
  File "/Users/will/src/opensource/ansible/bin/ansible-playbook", line 118, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python2.7/site-packages/ansible/cli/playbook.py", line 122, in run
    results = pbex.run()
  File "/usr/local/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 159, in run
    result = self._tqm.run(play=play)
  File "/usr/local/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 261, in run
    start_at_done=self._start_at_done,
  File "/usr/local/lib/python2.7/site-packages/ansible/executor/play_iterator.py", line 197, in __init__
    new_block = block.filter_tagged_tasks(play_context, all_vars)
  File "/usr/local/lib/python2.7/site-packages/ansible/playbook/block.py", line 388, in filter_tagged_tasks
    return evaluate_block(self)
  File "/usr/local/lib/python2.7/site-packages/ansible/playbook/block.py", line 383, in evaluate_block
    new_block.block = evaluate_and_append_task(block.block)
  File "/usr/local/lib/python2.7/site-packages/ansible/playbook/block.py", line 377, in evaluate_and_append_task
    task.evaluate_tags(play_context.only_tags, play_context.skip_tags, all_vars=all_vars)):
  File "/usr/local/lib/python2.7/site-packages/ansible/playbook/taggable.py", line 63, in evaluate_tags
    tags = set([i for i, _ in itertools.groupby(tags)])
TypeError: unhashable type: 'list'
```

**Comment by Akasurde** [comment]
cc @sivel

**Comment by bcoca** [comment]
so your use case confuses me, if you want to 'always' run the include_role, but internal tasks ONLY when they match tags, just 

```
- include_role: name=common
  tags: ['always']
```

The tags will not be inherited (it only applies to the include_role) and any tasks inside will be evaluated for their own tags.

Also, since you don't seem to be using anything dynamic, you can use `import_role` , which ignores tags itself and applies any of them to included tasks
```
 - import_role: name=common
```

**Comment by sivel** [comment]
This exception is not limited to `include_role`.  The exception provided above occurs even with a more simple playbook:

```yaml
- hosts: localhost
  gather_facts: false
  tasks:
    - debug:
        msg: common
      tags: "{{ ansible_run_tags | default('always') }}"
```

Results in:

```
Traceback (most recent call last):
  File "/Users/matt/projects/ansibledev/ansible/bin/ansible-playbook", line 118, in <module>
    exit_code = cli.run()
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/playbook.py", line 122, in run
    results = pbex.run()
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/executor/playbook_executor.py", line 159, in run
    result = self._tqm.run(play=play)
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/executor/task_queue_manager.py", line 261, in run
    start_at_done=self._start_at_done,
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/executor/play_iterator.py", line 197, in __init__
    new_block = block.filter_tagged_tasks(play_context, all_vars)
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/playbook/block.py", line 394, in filter_tagged_tasks
    return evaluate_block(self)
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/playbook/block.py", line 389, in evaluate_block
    new_block.block = evaluate_and_append_task(block.block)
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/playbook/block.py", line 383, in evaluate_and_append_task
    task.evaluate_tags(play_context.only_tags, play_context.skip_tags, all_vars=all_vars)):
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/playbook/taggable.py", line 63, in evaluate_tags
    tags = set([i for i, _ in itertools.groupby(tags)])
TypeError: unhashable type: 'list'
```

**Comment by daMupfel** [comment]
Any news on this issue? We ran into the same problem.

**Comment by sivel** [comment]
This issue was resolved in 7eb1ab45a7cdc4546795845fd56d4f6506db0234 and subsequently included in the 2.8 release.

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project
